internal/controller/eventing/integrationtests/controller/integration_test.go

```diff
@@ -627,30 +662,43 @@ func Test_DeleteEventingCR(t *testing.T) {
 			testEnvironment.EnsureDeploymentExists(t, eventing.GetPublisherDeploymentName(*tc.givenEventing), givenNamespace)
 			testEnvironment.EnsureHPAExists(t, eventing.GetPublisherDeploymentName(*tc.givenEventing), givenNamespace)
 
-			testEnvironment.EnsureEventingResourceDeletion(t, tc.givenEventing.Name, givenNamespace)
+			if tc.givenSubscription != nil {
+				// create subscriptions if given.
+				testEnvironment.EnsureNamespaceCreation(t, tc.givenSubscription.Namespace)
+				testEnvironment.EnsureK8sResourceCreated(t, tc.givenSubscription)
+				testEnvironment.EnsureSubscriptionExists(t, tc.givenSubscription.Name, tc.givenSubscription.Namespace)
 
-			// then
-			if *testEnvironment.EnvTestInstance.UseExistingCluster {
-				testEnvironment.EnsureDeploymentNotFound(t, eventing.GetPublisherDeploymentName(*tc.givenEventing), givenNamespace)
-				testEnvironment.EnsureHPANotFound(t, eventing.GetPublisherDeploymentName(*tc.givenEventing), givenNamespace)
-				testEnvironment.EnsureK8sServiceNotFound(t,
-					eventing.GetPublisherPublishServiceName(*tc.givenEventing), givenNamespace)
-				testEnvironment.EnsureK8sServiceNotFound(t,
-					eventing.GetPublisherMetricsServiceName(*tc.givenEventing), givenNamespace)
-				testEnvironment.EnsureK8sServiceNotFound(t,
-					eventing.GetPublisherHealthServiceName(*tc.givenEventing), givenNamespace)
-				testEnvironment.EnsureK8sServiceAccountNotFound(t,
-					eventing.GetPublisherServiceAccountName(*tc.givenEventing), givenNamespace)
+				//then
+				// givenSubscription existence means deletion of Eventing CR should be blocked.
+				testEnvironment.EnsureK8sResourceDeleted(t, tc.givenEventing)
+				testEnvironment.GetEventingAssert(g, tc.givenEventing).Should(tc.wantMatches)
 			} else {
-				// check if the owner reference is set.
-				// if owner reference is set then these resources would be garbage collected in real k8s cluster.
-				testEnvironment.EnsureEPPK8sResourcesHaveOwnerReference(t, *tc.givenEventing)
-				// ensure clusterrole and clusterrolebindings are deleted.
+				// then
```

Comment:
Where are you deleting the Eventing CR in the else block?

Reply:
it is deleted in the line 673 `testEnvironment.EnsureK8sResourceDeleted(t, tc.givenEventing)`. 
For the no subscription cases, it is deleted in the line 678.